warehouse/views.py

```diff
@@ -134,6 +134,22 @@ def forbidden(exc, request):
 
     # Check if the error has a "result" attribute and if it is a WarehouseDenied
     if hasattr(exc, "result") and isinstance(exc.result, WarehouseDenied):
+        # If the forbidden error is because the user does not have a verified
+        # email address, redirect them to their account page for email verification.
+        if exc.result.reason == "unverified_email":
+            request.session.flash(
+                request._(
+                    "You must verify your email address before you can perform "
+                    "this action."
```

Comment:
I think we can be more specific here: the _primary_ email address needs to be verified. It's possible the user has a verified email, but it's not their primary email.

Reply:
Excellent point, will reword!